examples/CampaignManagement/ValidateTextAd.php

```diff
@@ -0,0 +1,197 @@
+            'ad' => $ad
+        ]);
+
+        // Creates the ad group ad operation.
+        $adGroupAdOperation = new AdGroupAdOperation();
+        $adGroupAdOperation->setCreate($adGroupAd);
+        $adGroupAdServiceClient = $googleAdsClient->getAdGroupAdServiceClient();
+
+        try {
+            // Try sending a mutate request to add the ad group ad.
+            $response = $adGroupAdServiceClient->mutateAdGroupAds(
+                $customerId,
+                [$adGroupAdOperation],
+                ['validateOnly' => true]
+            );
+
+            // Since validation is ON, result will be null.
+            printf("Expanded text ad validated successfully.%s", PHP_EOL);
+        } catch (GoogleAdsException $googleAdsException) {
+            // This block will be hit if there is a validation error from the server.
+            printf("There were validation error(s) while adding expanded text ad.%s", PHP_EOL);
+
+            $count = 1;
+            foreach ($googleAdsException->getGoogleAdsFailure()->getErrors() as $googleAdsError) {
+                /** @var GoogleAdsError $googleAdsError */
+                if ($googleAdsError->getErrorCode()->getPolicyFindingError()
+                    == PolicyFindingError::POLICY_FINDING
+                    || $googleAdsError->getErrorCode()->getPolicyViolationError()
+                    == PolicyViolationError::POLICY_ERROR) {
+                    // Only one of PolicyFindingDetails or PolicyViolationDetails will be
+                    // populated. PolicyViolationDetails is used by some ad formats, and
+                    // PolicyFindingDetails by others.
+                    if ($googleAdsError->getDetails()->getPolicyFindingDetails()) {
+                        $details = $googleAdsError->getDetails()->getPolicyFindingDetails();
+                        foreach ($details->getPolicyTopicEntries() as $entry) {
+                            /** @var PolicyTopicEntry $entry */
+                            printf(
+                                "%d) Policy topic entry with topic '%s' and type '%s'" .
+                                " was found.%s",
+                                $count,
+                                $entry->getTopicUnwrapped(),
+                                PolicyTopicEntryType::name($entry->getType()),
+                                PHP_EOL
+                            );
+                        }
+                    } else if ($googleAdsError->getDetails()->getPolicyViolationDetails()) {
+                        $details = $googleAdsError->getDetails()->getPolicyViolationDetails();
+                        printf(
+                            "%d) Policy violation with name '%s' and isExemptable %s" .
+                            " was found.%s",
+                            $count,
+                            $details->getExternalPolicyName(),
+                            var_export($details->getIsExemptible(), true),
```

Comment:
Can you explain why we need var_export for this? To return True and False?

Reply:
Yes, do you know an elegant way of printing boolean value with printf?

Reply:
I think there is no one way to do that.
Previously, we usually use [this way](https://stackoverflow.com/a/2795180) though, which is personally a bit better because the code readers wouldn't need to know about var_export (and its "true" parameter).

Reply:
SGTM, updated